server/src/storage/datastore.ts

```diff
@@ -38,18 +40,23 @@ const extractAndAppendId = (res: Entity) => {
  * @param kind The Kind that is being queried
  * @param id The id of the Entity being queried
  */
-const getWithId = async (kind: string, id: string) => {
-  const key = datastore.key([kind, datastore.int(id)]);
+const getWithId = async (kind: string, id: number) => {
+  const key = datastore.key([kind, id]);
   const [res] = await datastore.get(key);
   return extractAndAppendId(res);
 };
 
 /**
  * A Datastore wrapper that gets all entities of a specified Kind.
  * @param kind The Kind that is being queried
+ * @param filters Any filters that will be applied to the query
  */
-const getAllWithId = async (kind: string) => {
-  const query = datastore.createQuery(kind);
+const getAllWithId = async (kind: string, filters?: Filter[]) => {
```

Comment:
I think we should rename this to `getAll`. `getWithId` means that you want to get all the objects having the specified id right? So this one should be `getAll` or `getAllWithFilter`. But since the filter is optional then just `getAll` is better.

Reply:
Oh okay I meant it like `WithId` meaning the thing you get comes with the `id`. Okay but since there's some confusion I will just rename both of them then HAHA